tools/hardware/outputs/json.py

```diff
@@ -0,0 +1,64 @@
+#
+# Copyright 2023, UNSW
+#
+# SPDX-License-Identifier: GPL-2.0-only
+#
+
+''' Generate a JSON file with memory region info from the device tree '''
+
+import argparse
+import json
+from typing import List
+import hardware
+from hardware.config import Config
+from hardware.fdt import FdtParser
+from hardware.utils.rule import HardwareYaml
+
+
+def make_json_list_of_regions(regions) -> List:
+    return [
+        {
+            'start': r.base,
+            'end':   r.base + r.size
+        }
+        for r in regions if r.size > 0
+    ]
```

Comment:
Should this be represented as a hexadecimal string instead of an int? Makes it more human readable (and would only add an extra step in whatever python script is being used, I assume)

Reply:
Unfortunately, JSON supports decimal integers only (see https://www.json.org/json-en.html). Since even comments are not allows in JSON, the only way to do this would be having a dummy string element like `"_comment_start": "..."`. But I'd prefer using a `"description": "<from> - <to>: foo42 device"` in this case.

There is an extension JSON5 (see https://json5.org/) that adds hex number and comment, but I wonder if we want to use this?

Reply:
> Should this be represented as a hexadecimal string instead of an int? Makes it more human readable (and would only add an extra step in whatever python script is being used, I assume)

Yes, that would be better since it's more consistent with `gen_config.json`.

Reply:
> I am aware, I meant is it not possible to put the hex in a string? "0x0" -> "0x1000" for example.

This would work. But it contradicts the intention of using JSON as machine processable standardized data interchange format, because you'd have to parse the string fields manually then. The "description" would be a trade off, it provides something for humans, but at the price of a certain redundancy - which seem acceptable, as no person is supposed to edit/create these JSON files manually.

Reply:
Done